bk-res-provider

kkocaburak opened the pull request:
-> add ResProvider class to provider resources
	-> refactor AndroidViewModel extension from BaseAndroidViewModel to ViewModel, refactor the name of BaseAndroidViewModel to BaseViewModel
	-> refactor application value in BaseAndroidViewModel constructor and subclasses
	-> add resProvider variable to BaseViewModel

Commit:
bk-res-provider
	-> add ResProvider class to provider resources
	-> refactor AndroidViewModel extension from BaseAndroidViewModel to ViewModel, refactor the name of BaseAndroidViewModel to BaseViewModel
	-> refactor application value in BaseAndroidViewModel constructor and subclasses
	-> add resProvider variable to BaseViewModel

hcknl commented on app/src/main/kotlin/com/adesso/movee/application/MainApplication.kt:
```diff
@@ -8,10 +10,34 @@ import timber.log.Timber
+            }
+
+            override fun onActivityResumed(activity: Activity) {
+                currentActivity = activity
```
uygulama arka plana alınınca currentActivity hala dolu olacak sıfırlanması gerekmiyor mu?

kkocaburak replied:
Evet burada yanlış bir bağımlılık oluşturmuştum, currentActivity değişkenini kaldırdım, teşekkürler.

muratbilir commented on app/src/main/kotlin/com/adesso/movee/base/ResProvider.kt:
```diff
@@ -0,0 +1,36 @@
+import com.adesso.movee.application.MainApplication
+import javax.inject.Inject
+
+class ResProvider @Inject constructor(val application: Application) {
```
Resource erişimi için application context kullanmak doğru değildir. Hem "theme" bilgisine sahip değil, hem de "configuration change" durumlarında yanlış veri verecektir. Burada "activity context" kullanılmalı ve bu sınıf "activity lifecycle scope" içerisinde yaşamalıdır. Bu yüzden viewModel içinde de kullanılamaz.

kkocaburak replied:
ResProvider class'ını silip yerine ResourceProvider class'ını inject ettim. ResourceProvider içerisinde şimdilik string çekme işlemleri yapıyorum. Activity context ile stringler çekilecek şekilde güncelledim, teşekkürler.

Commit:
bk-res-provider
	-> removed ResProvider class
	-> removed MainApplication's application dependency
	-> add injection of ResourceProvider class to BaseViewModel

ismailakkulakadesso reviewed (approved):
Mevcut resource provider kullanımını ilerleyen zamanlarda değiştirebiliriz ama bu haliyle iyi görünüyor.

Commit:
Merge branch 'develop' into feature/bk-res-provider

# Conflicts:
#	app/src/main/kotlin/com/adesso/movee/scene/moviedetail/MovieDetailViewModel.kt
